tools/vms-generator/utils/utils.go

```diff
@@ -642,7 +641,7 @@ func getBaseVM(name string, labels map[string]string) *v1.VirtualMachine {
 			Labels: labels,
 		},
 		Spec: v1.VirtualMachineSpec{
-			Running: &running,
+			RunStrategy: k6tpointer.P(v1.RunStrategyHalted),
```

Comment:
do you think i should replace usages of `k6tpointer "kubevirt.io/kubevirt/pkg/pointer"` and `"k8s.io/utils/pointer"` to be used with `ptr.To` everywhere in the file ? or it's out of the scope

Reply:
I think it would be out of the scope, would be awesome to see it in the following PR.

Reply:
@EdDev has a strong opinion about avoiding external dependencies, when we have simple implementation inside. Please keep `k6tpointer` unless you have a strong reason not to.

Reply:
@EdDev Why it's needed to have pointer package where in k8s's ptr package you have the exact same implementation? Shouldn't we get rid of this package?

```
// pkg/pointer
func P[T any](t T) *T {
	return &t
}

// k8s.io/utils/ptr
func To[T any](v T) *T {
	return &v
}
```

By the way, what does `k6t` mean in `k6tpointer`? Can you just name it `pointer`?

Reply:
> By the way, what does k6t mean in k6tpointer? Can you just name it pointer?

k6t is an abbreviation for KubeVirt like K8s is an abbreviation for Kubernetes.

Reply:
@codingben Hi, we always strive to reduce dependency on external packages unless its really needed, especially those that introduce small self-contained toolbox for your project. Main reason is that its easier to maintain, you don't to be dependent on external project policies.

Reply:
> @EdDev has a strong opinion about avoiding external dependencies, when we have simple implementation inside. Please keep `k6tpointer` unless you have a strong reason not to.

+1 to not adding dependencies. I'm with @EdDev on this one.

> Why it's needed to have pointer package where in k8s's ptr package you have the exact same implementation?

Just because the implementation is a one-liner (or 3? :smile:) it does not worth bringing in another dependency IMO. Also, in this case we prefer to avoid third part updates to this package to keep it as simple as it is now, unless there's a strong reason to do otherwise.

Reply:
@enp0s3 @iholder101 Thanks for replying. I respect you but I disagree with you guys. It's just my opinion and you can approve it without my approval of course, I'm not even an approver or reviewer here.

----

Since we're using Kubernetes objects, to me it also makes sense to use any other utils provded by Kubernetes library.

I see that `pointer` package in KubeVirt was introduced in Dec 2022 with main one reason, according to Eddy [1]:
```
It is aimed to replace `k8s.io/utils/pointer` with a generic version.
Unlike the previous version, the generic one accepts any type.
```
Because before you had this:
```
pointer.Bool(secureBoot)
```
And with Go 1.18 that added generics support you can do:
```
pointer.P(secureBoot)
```

IMO, this is the real reason that `pointer` package is there until today. IMO, it's no longer needed because it can be replaced by `k8s.io/utils/ptr` that supports generic types.

[1] https://github.com/kubevirt/kubevirt/commit/7b7a6fd86dbe57bbadcb09d25bdbf3be1d3291cf

Reply:
Just a final note: This dependency `k8s.io/utils/pointer` is already used widely in KubeVirt codebase. Someone needs to convince contributors not to use `k8s.io/utils/ptr` instead in the future.

Reply:
> @enp0s3 @iholder101 Thanks for replying. I respect you but I disagree with you guys. It's just my opinion and you can approve it without my approval of course, I'm not even an approver or reviewer here.

> to me it also makes sense to use any other utils provded by Kubernetes library.

Thanks for the discussion @codingben!

It takes courage to share your opinion as an underdog. I deeply encourage you to always share your honest opinion, as it's perfectly okay (and even beneficial!) to disagree :)

It makes sense to rely on external dependencies sometimes, and we definitely do that in Kubevirt. However, there's a cost for external dependencies that we have to consider, and there's always a trade-off.

I respect you and your opinion, but in the end of the day we have to make a choice. In this specific case there seems to be a consensus around not bringing in this external dependency. Since I saw there's a large consensus for it, I've approved the PR.

Thanks again for this discussion and for your honest opinion!

Reply:
> Just a final note: This dependency `k8s.io/utils/pointer` is already used widely in KubeVirt codebase, it's deprecated and suggested by Kubernetes maintainers to be replaced by `k8s.io/utils/ptr`.

You're right. These usages predate the current Kubevirt pointer package, and we should definitely drop this dependency moving forward.

Reply:
> It takes courage to share your opinion as an underdog.

I disagree. It's just another request that came from OpenShift Virtualization management, especially from @dankenigsberg and @EdDev to review code cleanup PRs in kubevirt/kubevirt repository. From the beginning it felt wrong - exactly like "underdog".

cc @dominikholler

Reply:
By "underdog" I meant that most of the people that took part in the discussion had the other opinion.
As I've said above, I deeply encourage you to always share your honest opinion even if there's disagreement.